src/dual-kad-dht.ts

```diff
@@ -81,6 +83,36 @@ class DHTPeerRouting implements PeerRouting {
   }
 }
 
+function multiaddrIsPublic (multiaddr: Multiaddr): boolean {
+  const tuples = multiaddr.stringTuples()
+
+  // p2p-circuit should not enable server mode
+  for (const tuple of tuples) {
```

Comment:
Can we use the [protoCodes](https://github.com/multiformats/js-multiaddr/blob/8582c09da7f601709f843a195286cd6b33a3a201/src/index.ts#L593) function to do these kinds of checks? so something like

``` typescript

const P2P_CIRCUIT_CODE = 290
const DNS4_CODE = 54
// rest of other codes...
function multiaddrIsPublic (multiaddr: Multiaddr): boolean {
	multiaddr.forEach((maddr) => {
		 if (maddr.protoCodes().includes(CIRCUIT_RELAY_CODE)) {
			return false
		 }
		 
		 if (maddr.protoCodes().includes(DNS4_CODE) {
			return true
		}
	})

 	// other checks
 	
 	return false
}

```

Reply:
Calling `protoCodes()` repeatedly like this is expensive.

I've swapped the magic numbers out for named constants instead, hopefully that's a bit clearer.